### PR TITLE
Refactor and UI revamp

### PR DESCRIPTION
**Pull request by azymohliad**
UI changes:
- Lexemes UI:
    - Tweaked the layout and text formatting.
    - Added language level and rection, removed tags.
    - Made radio button auto-hide if there's only one lexeme.
    - Implemented lexeme selection by clicking anywhere on the `LexemeWidget`.
    - Changed "Näited" label to "Examples" - it would be nice to make the whole UI translatable some day, but for now it is in English, so I'd prefer to keep it consistent. Card templates aren't part of the addon UI, so there's no problem with "Näited" there.
- Header bar:
    - Added focus mnemonics shortcuts (`Alt+<Letter>` to focus an element).
    - Disabled dictionary selector while search is in progress.
    - Disabled language selector while google translation is in progress.
    - Tweaked header bar element size policies.
- Other:
    - Disabled search bar while search is in progress - otherwise pressing enter can trigger a new search, which causes the results to be displayed multiple times.
    
image

Non-UI changes in the UI module:
- Fixed google translations not saved to notes - a regression from #2.
- Simplified note existence and equality checks.
- Moved translation logic into `LexemeWidget`.
- Fixed crash when translation response arrives after widget is deleted (similar to #4, but for translations).

Sonaveeb module changes:
- Renamed `WordInfo.pos` to `WordInfo.word_class` - PoS acronym is ambiguous.
- Removed lexeme examples limit - it is more appropriate to apply in the UI layer.
- Added language level validation - it is more appropriate here than in the UI layer.
- Added timeout argument for all web requests - otherwise, hanging request may hang Anki even when trying to close it. A better solution would be to implement manual cancellation, but it looks unreasonably complicated with `requests`, so maybe later.
- Removed exception handling from `select_dictionary` - I think it should propagate the error honestly here.
- Tweaked dictionaries' names and descriptions to be a little more descriptive.
- Changed the default dictionary to `lite` - my hunch is that this is what most users would prefer.

Google translate module changes:
- Added docs.
- Added timeout argument.
- Fixed test script.

### CODE REVIEW

File: anki_addon/sonaveeb.py [hunk 1]
```diff
@@ -100,10 +103,11 @@ def _parse_lexeme_definition(self, definition_row):
...
         # Extract language level if present
-        for meta_span in definition_row.find_all(class_='additional-meta'):
-            if meta_span.get('title') == 'Keeleoskustase':
-                level = meta_span.string.strip()
-                break
+        if level := definition_row.find(class_='additional-meta', title='Keeleoskustase'):
+            level = level.string.strip()
+            # TODO: Is this check necessary? Are there words with invalid level value?
+            if level not in VALID_LANGUAGE_LEVELS:
+                level = None
```
Comment (azymohliad) [inline anki_addon/sonaveeb.py:110]: Is checking against `VALID_LANGUAGE_LEVELS` necessary? Have you seen words with invalid level values?

File: anki_addon/ui/lexeme.py [hunk 2]
```diff
@@ -2,148 +2,263 @@
...
+            examples_label.setWordWrap(True)
+            self.layout.addWidget(examples_label)
 
-        self.setLayout(self.layout)
+        # Add rection
+        if lexeme.rection:
+            rection = ', '.join(lexeme.rection)
+            rection_label = QLabel(f'**Rection:** {rection}')
+            rection_label.setTextFormat(Qt.TextFormat.MarkdownText)
+            self.layout.addWidget(rection_label)
```
Comment (azymohliad) [inline anki_addon/ui/lexeme.py:78]: So I've added rection to the `LexemeWidget`, do you think there's any point to display other lexeme.tags there? When you added tags to `LexemeWidget` before, was it primarily to display the rection, or is there anything else useful there?

File: anki_addon/ui/main_window.py [hunk 3]
```diff
@@ -23,9 +24,11 @@ def __init__(self, sonaveeb=None, parent=None):
         self._deck_selector = QComboBox()
         for deck in mw.col.decks.all_names_and_ids():
             self._deck_selector.addItem(deck.name, userData=deck.id)
-        self._deck_selector.currentIndexChanged.connect(self.deck_changed)
-        self._deck_selector.setMinimumWidth(300)
-        self._deck_selector.setSizePolicy(QSizePolicy.Policy.Expanding, QSizePolicy.Policy.Fixed)
+        self._deck_selector.currentIndexChanged.connect(self._on_deck_changed)
+        self._deck_selector.setMinimumWidth(100)
+        self._deck_selector.setMaximumWidth(500)
```
Comment (azymohliad) [inline anki_addon/ui/main_window.py:29]: I've changed width limits on deck selector. When you added min limit originally, was it to reduce the default for very long deck names, or increase it for very short ones? I didn't like how empty it looks when all deck names are very short, so I reduced it here.

**Review by azymohliad — COMMENTED** [review]
Hey @s10, before I merge this, could you help me with the questions below whenever you have time?

Also, let me know if you have any other concerns about this PR.

File: anki_addon/sonaveeb.py (hunk 1 above)
Comment (s10) [reply anki_addon/sonaveeb.py:112]: It's not necessary; I've never seen other values here.

File: anki_addon/ui/lexeme.py (hunk 2 above)
Comment (s10) [reply anki_addon/ui/lexeme.py:78]: There is no point in displaying other tags. I'm not even sure that they exist at all. The only needed things are language level, rection and word class. Previously, all of them were under the "tags" category. I still believe that the best place for language level and word class is still in tags inside notes, not in separate fields.

File: anki_addon/ui/main_window.py (hunk 3 above)
Comment (s10) [reply anki_addon/ui/main_window.py:29]: I use some nested decks with extremely long names, like "Advanced Grammar in Use::053-059. Relative Clauses and Other Types Of Clause::Unit 056: Other ways of adding information to noun phrases 1: additional noun phrases, etc.", so I applied that limit to make that window look nicer.

File: anki_addon/ui/lexeme.py (hunk 2 above)
Comment (azymohliad) [reply anki_addon/ui/lexeme.py:78]: Ah, no, I'm not adding word class and language level as separate note fields, they are note tags as before. I was only asking about `LexemeWidget` UI. But it answers my question, thanks

File: anki_addon/ui/main_window.py (hunk 3 above)
Comment (azymohliad) [reply anki_addon/ui/main_window.py:29]: Ok, then the new limits should handle that well too

**Comment by azymohliad** [comment]
I'm also thinking to rename the concept of "Dictionary" into "Sõnaveeb Mode" or just "Mode" for short. The Sõnaveeb documentation describes them as modes [here](https://sonaveeb.ee/about?uilang=en#det):
> The user can currently choose between two modes of information display: Sõnaveeb (for advanced users) or Learner's Sõnaveeb (for language learners).

And the word "dictionary" is used there for other things:
> Sõnaveeb [WordWeb] is the language portal of the Institute of the Estonian Language (EKI) containing lexical information from a growing number of **dictionaries** and databases.

So I'm afraid the word "dictionary" in the addon may be misleading.

**Pull request merged**